waltz-model/src/main/java/com/khartec/waltz/model/EntityReference.java

```diff
@@ -73,5 +74,14 @@ public static EntityReference mkRef(EntityKind kind, long id, String name, Strin
     }
 
 
+    public static EntityReference mkRef(Map map) {
+        return EntityReference.mkRef(
+                Enum.valueOf(EntityKind.class, map.get("kind").toString()),
+                Long.parseLong(map.get("id").toString()),
+                (String) map.get("name"),
+                (String) map.get("description"));
+    }
+
```

Comment:
Just a q:  does this do the right thing when `name` and `description` are null ?   Perhaps add some unit tests... ;)

Reply:
agreed, unit tests included